File: src/main/java/com/github/xumeng/java/problems/easy/problem0326/Solution0326.java
package com.github.xumeng.java.problems.easy.problem0326;

public class Solution0326 {

    public boolean isPowerOfThree(int n) {
        if (n < 1) {
            return false;
        }
        int base = 3;
        double power = 0.0;
        for (int i = 0; power <= Integer.MAX_VALUE; i++) {
            power = Math.pow(base, i);
            if (n == power) {
                return true;
            }
            if (n < power) {
                return false;
            }
        }
        return false;
    }

}

File: src/main/kotlin/com/github/xumeng/kotlin/problems/easy/problem0326/Solution0326.kt
package com.github.xumeng.kotlin.problems.easy.problem0326

import kotlin.math.pow

class Solution0326 {

    fun isPowerOfThree(n: Int): Boolean {
        if (n < 1) {
            return false
        }
        val base = 3.0
        var i = 0
        var power = base.pow(i)
        while (power <= Int.MAX_VALUE) {
            if (n == power.toInt()) {
                return true
            }
            if (n < power) {
                return false
            }
            i++
            power = base.pow(i)
        }
        return false
    }

}